src/pages/Home/components/TodoCreator/TodoCreator.tsx

```diff
@@ -1,66 +1,98 @@
-import { ChangeEvent, useState } from 'react';
+// import { ChangeEvent, useState } from 'react';
 
+import { useForm } from 'react-hook-form';
 import { useDispatch, useSelector } from 'react-redux';
 
 import Button from '@mui/material/Button';
 import Typography from '@mui/material/Typography';
 import CircularProgress from '@mui/material/CircularProgress';
+import Tooltip from '@mui/material/Tooltip';
 
 import { AppDispatch, RootState } from 'src/redux/store';
 import { createTask } from 'src/redux/tasks';
 
 // import { useTodoContext } from '../../context/Todo';
 
 import { StyledOutlinedInput } from './styled';
+
+type FormData = {
+  task: string;
+};
+
 export default function TodoCreator() {
   // const { createTodo } = useTodoContext();
   const dispatch = useDispatch<AppDispatch>();
   const { isCreating } = useSelector((state: RootState) => state.tasks);
 
-  const [text, setText] = useState('');
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    resetField,
+  } = useForm<FormData>();
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setText(event.target.value);
+  const onSubmit = (data: FormData) => {
```

Comment:
We do have different naming rules for the functions, if it is conflicts with another function, you could rename it's name, eg `handleSubmit: handleRHFSubmit` or `handleSubmit: onSubmit`